src/Statistics.jl

```diff
@@ -197,9 +197,9 @@ if !isdefined(Base, :mean)
         end
     end
 
-    function mean(r::AbstractRange{<:Real})
-        isempty(r) && return oftype((first(r) + last(r)) / 2, NaN)
-        (first(r) + last(r)) / 2
+    function mean(r::AbstractRange{T}) where T
+        isempty(r) && return zero(T) / 0
+        return first(r) / 2 + last(r) / 2
```

Comment:
Probably more readable this way?
```suggestion
        isempty(r) && return zero(T)/0
        return first(r)/2 + last(r)/2
```